### PR TITLE
docs: Add missing alt text to gallery images for accessibility.

### PR DESCRIPTION
**Pull request by Sandeep3084**

### CODE REVIEW

File: docs/gallery.rst [hunk 1]
```diff
@@ -291,7 +291,7 @@ Examples that make use of the :doc:`api/contrib` module.
 .. only:: html
 
   .. image:: /images/examples/contrib/ademamix_rosenbrock.png
-    :alt:
+    :alt: AdEMAMix
```
Comment (selamw1) [inline docs/gallery.rst]: Thanks! LGTM!

Minor adjustment for consistency:
`:alt: AdEMAMix.`
_(Added a period to match the other entries)_

Comment (Sandeep3084) [reply docs/gallery.rst]: Got it. Thank you!

**Review by rdyro — COMMENTED** [review]
Hey, thanks for the PR!

In the spirit of accessibility I believe we'd need this alt-text to describe what's in the figure. The current alt-text isn't too helpful.

Would you be willing to add about 2 sentences for each figure describing what the figure is presenting? (e.g., "Loss landscape chart with the path taken by the L-BFGS optimizer overlaid on top." for lbfgs.png)

**Comment by Sandeep3084** [comment]
Hey @rdyro. I have added a detailed description for all the alt tags in the gallery, please look into it and let me know if you need anything else. Happy New Year!

File: docs/gallery.rst [hunk 2]
```diff
@@ -108,7 +108,7 @@
 .. only:: html
 
   .. image:: /images/examples/lookahead.png
-    :alt:
+    :alt: Accuracy comparison chart with the Lookahead wrapper achieving higher test accuracy faster than the standard Adam optimizer on MNIST.
```
Comment (rdyro) [inline docs/gallery.rst]: This description isn't correct.

File: docs/gallery.rst [hunk 3]
```diff
@@ -158,7 +158,7 @@
 .. only:: html
 
   .. image:: /images/examples/tiny_shakespeare.png
-     :alt: Character-level Transformer on Tiny Shakespeare
+     :alt: Screenshot of Shakespearean dialogue text generated by the character-level Transformer model.
```
Comment (rdyro) [inline docs/gallery.rst:161]: it'd be good to add "trained on the Tiny Shakespear dataset."

File: docs/gallery.rst [hunk 4]
```diff
@@ -192,7 +192,7 @@
 .. only:: html
 
   .. image:: /images/examples/perturbations.png
-    :alt:
+    :alt: Geometric diagram illustrating differentiable perturbations, showing a hexagonal constraint set C with vectors indicating the perturbed parameters and their projections.
```
Comment (rdyro) [inline docs/gallery.rst:195]: I'm not sure this is right, I'd go with: "A perturbed maximizer illustration from https://arxiv.org/abs/2002.08676"

File: docs/gallery.rst [hunk 5]
```diff
@@ -258,7 +258,7 @@ Examples that make use of the :doc:`api/contrib` module.
 .. only:: html
 
   .. image:: /images/examples/contrib/reduce_on_plateau.png
-    :alt:
+    :alt: Learning rate schedule chart showing the learning rate value stepping down in discrete intervals over 50 epochs as the training progresses.
```
Comment (rdyro) [inline docs/gallery.rst:261]: It'd be good to emphasize the point about the learning rate being algorithmically adjusted; it should be more like "A chart showing an automatically decreasing learning rate via the Reduce on Plateau learning rate scheduler."

File: docs/gallery.rst [hunk 6]
```diff
@@ -275,7 +275,7 @@ Examples that make use of the :doc:`api/contrib` module.
 .. only:: html
 
   .. image:: /images/examples/contrib/sam.png
-    :alt:
+    :alt: Optimization trajectory plot comparing the path of a standard optimizer versus SAM, showing how SAM navigates the loss landscape to find flatter minima.
```
Comment (rdyro) [inline docs/gallery.rst:278]: nit: replace "standard" with "non-SAM" for a more accurate description.
